redhat-marketplace/redhat-marketplace-operator
# Feature/native deployments

**Changes Made:**

- removes the razee-deploy job and applies the watch-keeper deployment and remote-resourcesS3 controller from the operator's razeedeployment controller.
- creates an api for the parent rrs3 and associated types file
- uninstalls legacy resources created by the old razee-deploy-job
- alter the previous `fullUninstall()` func to only remove the razee related resources created by this pr
- the `rhm-operator-secret` is no longer need the `FILE_SOURCE_URL` field

**How to test:**
**1** .Install legacy resources
- `git checkout master`
- `make install`
- apply rhm-operator-secret

**2** .Run the image off this branch:
- `git checkout feature/native-deployments`
-  `make build`, `make push`
- `make migrate`

**3** .Ensure that the following legacy resources get deleted:
_CustomResources_
- RemoteResource
- FeatureFlagSetLD
- ManagedSet
- MustacheTemplate
- RemoteResourceS3Decrypt

_ClusterRole_
- razeedeploy-admin-cr
- redhat-marketplace-razeedeploy

_Service accounts_
- razeedeploy-sa
- watch-keeper-sa

_Deployments_
- watch-keeper
- clustersubscription
- featureflagsetld-controller
- managedset-controller
- mustachetemplate-controller
- remoteresource-controller
- remoteresources3-controller
- remoteresources3decrypt-controller

**4** .Ensure that the deployments `rhm-watch-keeper` and `rhm-remoteresources3-controller` deployments get created
